test/metabase/api/geojson_test.clj

```diff
@@ -67,6 +67,23 @@
             (is (valid? geojson) (str url))
             (is (thrown? clojure.lang.ExceptionInfo (valid? geojson)) (str url))))))))
 
+(deftest custom-geojson-disallow-overriding-builtins-test
+  (testing "We shouldn't let people override the builtin GeoJSON and put weird stuff in there; ignore changes to them"
+    (mt/with-temporary-setting-values [custom-geojson nil]
+      (let [built-in @#'geojson-api/builtin-geojson]
+        (testing "Make sure the built-in entries still look like what we expect so our test still makes sense."
+          (is (schema= {:us_states {:name     (s/eq "United States")
+                                    s/Keyword s/Any}
+                        s/Keyword  s/Any}
+                       built-in))
+          (is (= built-in
+                 (geojson-api/custom-geojson))))
+        (testing "Try to change one of the built-in entries..."
+          (geojson-api/custom-geojson (assoc-in built-in [:us_states :name] "USA"))
+          (testing "Value should not have actually changed"
+            (is (= built-in
+                   (geojson-api/custom-geojson)))))))))
```

Comment:
I added a test to be extra sure we're not letting people override the built-in keys to get around validation or anything like that